provider.go

```diff
@@ -25,90 +52,124 @@ type provider struct {
 	appClient    api.ApplicationsClient
 	spClient     api.ServicePrincipalClient
 	groupsClient api.GroupsClient
-	raClient     *authorization.RoleAssignmentsClient
-	rdClient     *authorization.RoleDefinitionsClient
+	raClient     *armauthorization.RoleAssignmentsClient
+	rdClient     *armauthorization.RoleDefinitionsClient
 }
 
 // newAzureProvider creates an azureProvider, backed by Azure client objects for underlying services.
-func newAzureProvider(settings *clientSettings, passwords api.Passwords) (api.AzureProvider, error) {
-	// build clients that use the GraphRBAC endpoint
-	userAgent := useragent.PluginString(settings.PluginEnv, userAgentPluginName)
-
-	var appClient api.ApplicationsClient
-	var groupsClient api.GroupsClient
-	var spClient api.ServicePrincipalClient
+func newAzureProvider(settings *clientSettings, passwords api.Passwords) (AzureProvider, error) {
+	httpClient := cleanhttp.DefaultClient()
 
-	graphURI, err := api.GetGraphURI(settings.Environment.Name)
+	cred, err := getTokenCredential(settings)
 	if err != nil {
 		return nil, err
 	}
 
-	graphApiAuthorizer, err := getAuthorizer(settings, graphURI)
+	msGraphAppClient, err := api.NewMSGraphApplicationClient(settings.GraphURI, cred)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to create MS graph client: %w", err)
 	}
 
-	msGraphAppClient, err := api.NewMSGraphApplicationClient(settings.SubscriptionID, userAgent, graphURI, graphApiAuthorizer)
+	opts := getClientOptions(settings, httpClient)
+
+	raClient, err := armauthorization.NewRoleAssignmentsClient(settings.SubscriptionID, cred, opts)
 	if err != nil {
 		return nil, err
 	}
 
-	appClient = msGraphAppClient
-	groupsClient = msGraphAppClient
-	spClient = msGraphAppClient
-
-	// build clients that use the Resource Manager endpoint
-	resourceManagerAuthorizer, err := getAuthorizer(settings, settings.Environment.ResourceManagerEndpoint)
+	rdClient, err := armauthorization.NewRoleDefinitionsClient(cred, opts)
 	if err != nil {
 		return nil, err
 	}
 
-	raClient := authorization.NewRoleAssignmentsClientWithBaseURI(settings.Environment.ResourceManagerEndpoint, settings.SubscriptionID)
-	raClient.Authorizer = resourceManagerAuthorizer
-	raClient.AddToUserAgent(userAgent)
-
-	rdClient := authorization.NewRoleDefinitionsClientWithBaseURI(settings.Environment.ResourceManagerEndpoint, settings.SubscriptionID)
-	rdClient.Authorizer = resourceManagerAuthorizer
-	rdClient.AddToUserAgent(userAgent)
-
 	p := &provider{
 		settings: settings,
```

Comment:
Sorry another one unrelated but might be good to clean up while it's fresh 😄  We're keeping the `clientSettings` on the `azureSecretBackend`, `client`, and `provider`. I think we can safely remove it from `provider` since it's set but never used again.

Reply:
Ah nice, removed